src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,446 @@
+                            valid = false;
+                            LOG.error("processAmazonAccountDetails() there waw a problem building a DI for " +
+                                amazonAccountDetail.toString(), e);
+                            GlobalVariables.getMessageMap().getErrorMessages().clear();
+                        }
+                    }
+                }
+            }
+            count ++;
+        }
+        LOG.info("Processed " + count + " AWS records");
+        return valid;
+    }
+    
+    private Account determineWhichAccountToUse(String costCenter, String kfsAccountString) {
+        Account costAccount = buildAccount(costCenter);
+        if (validAccount(costAccount)) {
+            return costAccount;
+        }
+        
+        Account kfsAccount = buildAccount(kfsAccountString);
+        if (validAccount(kfsAccount)) {
+            return kfsAccount;
+        }
+        LOG.error("determineWhichAccountToUse() unable to find an active account, both cost account " + costCenter + " and kfs account " + kfsAccountString + " are invalid");
+        return null;
+    }
+    
+    private Account buildAccount(String accountNumber) {
+        if (StringUtils.isNotBlank(accountNumber)) {
+            try {
+                Account account = getAccountService().getUniqueAccountForAccountNumber(accountNumber);
+                return account;
+            } catch (Exception e) {
+                LOG.debug(accountNumber + " is an invalid account");
+            }
+        }
+        return null;
+    }
+    
+    private boolean validAccount(Account account) {
+        return account != null && account.isActive() && !account.isExpired();
+    }
+    
+    private boolean shouldBuildDistributionOfIncomeDocument(AmazonAccountDetail amazonAccountDetail, KualiDecimal cost) {
+        for (DocumentHeader dh : findDocumentHeadersForAmazonDetail(amazonAccountDetail)) {
+            try {
+                DistributionOfIncomeAndExpenseDocument di = (DistributionOfIncomeAndExpenseDocument) getDocumentService().getByDocumentHeaderId(dh.getDocumentNumber());
+                if (di.getTotalDollarAmount().equals(cost)) {
+                    LOG.info("shouldBuildDistributionOfIncomeDocument() DI document number " + di.getDocumentNumber() + " already exists for Amazon account: " + amazonAccountDetail.toString());
+                    return false;
+                }
+            } catch (WorkflowException e) {
+                LOG.error("shouldBuildDistributionOfIncomeDocument() There was a problems creating a DI document", e); 
+                throw new RuntimeException(e);
+            }
+        }
+        return true;
+    }
+    
+    private Collection<DocumentHeader> findDocumentHeadersForAmazonDetail(AmazonAccountDetail amazonAccountDetail) {
+        Map fieldValues = new HashMap();
+        fieldValues.put("EXPLANATION", buildDocumentExplanation(amazonAccountDetail.getAwsAccount()));
```

Comment:
Do you need to have period in the criteria to only find documents created for the same period you're trying to create the DI for now?